docs/core-features/non_linear_operations.md

```diff
@@ -1,31 +1,32 @@
 
 ## Changing bit width in the MLIR or dynamically with a TLU
 
 Often, for binary operations, we need to have equivalent bit width for the two operands: it can be done in two ways. Either directly in the MLIR, or dynamically (i.e., at execution time) with a TLU. Because of these different methods, and the fact that none is stricly better than the other one in the general case, we offer different configurations for the non-linear functions.
 
-The first method has the advantage to not require an expensive TLU. However, it may have impact in other parts of the program, since the operand of which we change the bit width may be used elsewhere in the program, so it may create more bit widths changes. Also, if ever the modified operands are used in TLUs, the impact may be significative.
+Binary operations often require operands to have matching bit widths. This adjustment can be achieved in two ways: either directly within the MLIR or dynamically at runtime using a TLU. Each method has its own advantages and trade-offs, so Concrete provides multiple configuration options for non-linear functions.
 
-The second method has the advantage to be very local: it has no impact elsewhere. However, it is costly, since it uses a TLU.
+**MLIR adjustment:** This method doesn't require an expensive TLU. However, it may affect other parts of your program if the adjusted operand is used elsewhere, potentially causing more changes.
 
-## Generic Principle for the user
+**Dynamic adjustment with TLU:** This method is more localized and won’t impact other parts of your program, but it’s more expensive due to the cost of using a TLU.
 
-In the following non-linear operations, we propose a certain number of configurations, using the two methods on the different operands. In general, it is not easy to know in advance which configuration will be the fastest one, but with some Concrete experience. We recommend the users to test and try what are the best configuration depending on their circuits.
+## General guidelines
 
-By running the following programs with `show_mlir=True`, the advanced user may look the MLIR, and see the different uses of TLUs, bit width changes in the MLIR and dynamic change of the bit width. However, for the classical user, it is not critical to understand the different flavours. We would just recommend to try the different configurations and see which one fits the best for your case.
+In the following non-linear operations, we propose a certain number of configurations, using the two methods on the different operands. It’s not always clear which option will be the fastest, so we recommend trying out different configurations to see what works best for your circuit.
+
+Note that you have the option to set `show_mlir=True` to view how the MLIR handles TLUs and bit width changes. However, it's not essential to understand these details. so we recommend just testing the configurations and pick the one that performs best for your case.
```

Comment:
"However, it's not essential to understand these details. so we recommend": there is a miss capitalization to the 'so', or a punctuation problem